pandas/tests/io/test_sql.py

```diff
@@ -77,6 +77,9 @@
     SQLALCHEMY_INSTALLED = False
 
 
+pytestmark = [pytest.mark.db, pytest.mark.single_cpu]
```

Comment:
I think this shouldn't necessary (except `pytest.mark.db` used for mysql/postgres)

1. The sqllite3 engine should be able to run without the `pytest.mark.db` marker since it uses `:memory:` I believe i.e. these tests shouldn't be skipped in the CI if `no db` was specified
2. IIRC the xdist parallelization is done per file, so tests here should be run sequentially

Reply:
So are you thinking we should only mark the SQLAlchemy tests as db? Or none at all??

Reply:
Yeah I think the sqlalchemy tests and maybe wrap the fixture params that use sqlalchemy connections/engines in `pytest.param(..., marks=pytest.mark.db)`